addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2368,6 +2368,70 @@ var SnippetsMenu = Widget.extend({
     _onSnippetSearchResetClick: function () {
         this._filterSnippets('');
     },
+    /**
+     * Adds the computation to the mutex queue and sets a loading effect over the editor after 1s.
+     * As soon as the mutex is unlocked, the loading effect will be removed.
+     * if the '_setLoadingEffect' has not already been executed, the 'mutexPromise' will be able
+     * to stop the execution by calling clearTimeout().
+     *
+     * @private
+     * @param {OdooEvent} ev
+     */
+    _execSnippetEditionRequest: async function (ev) {
+        this._mutex.exec(ev.data.exec);
+        const loadingTimer = setTimeout(() => {
+            this._setLoadingEffect();
+        }, 1000);
+        const mutexPromise = (timer) => {
+            return this._mutex.getUnlockedDef().then(() => {
+                clearTimeout(timer);
+                this._removeLoadingEffect();
+            });
+        };
+        await mutexPromise(loadingTimer);
+    },
+    /**
+     * @private
+     */
+    _setLoadingEffect: function () {
+        const noActiveLoader = this.el.querySelector('.oe_option_tab_loader') === null;
+        if (noActiveLoader) {
+            const loaderContainer = document.createElement('div');
+            const loader = document.createElement('i');
+            const loaderContainerClassList = [
+                'd-flex',
+                'h-100',
+                'w-100',
+                'bg-black-25',
+                'position-absolute',
+                'oe_option_tab_loader',
+            ];
+            const loaderClassList = [
+                'fa',
+                'fa-spinner',
+                'fa-pulse',
+                'fa-4x',
+                'w-100',
+                'text-white',
+                'text-center',
+                'position-absolute',
+                'align-self-center',
+            ];
+            loaderContainer.classList.add(...loaderContainerClassList);
+            loader.classList.add(...loaderClassList);
+            loaderContainer.appendChild(loader);
+            this.el.appendChild(loaderContainer);
+        }
+    },
+    /**
+     * @private
+     */
+    _removeLoadingEffect: function () {
+        const loader = this.el.querySelector('.oe_option_tab_loader');
+        if (loader !== null) {
+            this.el.removeChild(loader);
+        }
+    },
```

Comment:
Thos methods are private method which are not handles, they should be in the private section, not the handlers one